### DISCUSSION TITLE
DrawXBitmap issue/confusion

### DISCUSSION THREAD

**Discussion by sparx266** [Bodmer/TFT_eSPI#2181] [General]
OK, I’m trying to use this sketch:

https://github.com/CelliesProjects/moonPhase-esp32/blob/master/examples/esp32/ILI9341/ILI9341.ino

But with TFT_eSPI instead of Adafruit_ILI9341.

The pictures are stored as RGB565 (16-bit) but TFT_eSPI expects an 8 bit value here.
Is this correct?
What is the best way around this please?

Thanks and regards.

**Comment by Bodmer** [comment]
The library can handle that image type with the pushImage function. Just replace the functions so instead of:

tft.drawRGBBitmap( 100, 20, (uint16_t *)phase_000.pixel_data, 80, 80);
use:
tft.pushImage( 100, 20, 80, 80, (uint16_t *)phase_000.pixel_data);

**Comment by sparx266** [comment]
Thanks Bodmer, but that didn't quite do it.
The image is displayed but its not as clear as in the original code.
Is there anything else that could be causing this issue?

Regards

**Reply by Bodmer** [reply]
I expect the colours are "byte swapped".
Add this line to the end of setup():

```
tft.setSwapBytes(true);
```

**Comment by sparx266** [comment]
That's got it.

Thank you so much.
